amanuensis/resources/userdatamodel/userdatamodel_state.py

```diff
@@ -52,13 +60,32 @@ def get_state_by_code(current_session, code):
 def get_all_states(current_session):
     return current_session.query(State).all()
 
+def notify_user_data_delivered(request, current_session):
+    """
+    Notify the users when project state changes to DATA_DELIVERED.
+    """
+    project = get_project_by_id(current_session, request.project_id)
+    email_subject = f"Project {project.name}: Data Delivered"
+    email_body = f"The project f{project.name} data was delivered."
+
+    return send_message(current_user.id, request.id, email_subject, email_body)
 
 def update_project_state(current_session, requests, state):
+    '''
+    Updates the state for a project, including all requests in the project. Notifies users when the state changes to DATA_DELIVERED.
+    '''
+
     for request in requests:
-        request.states.append(state)
+        if request.request_has_state[-1].state_id == 2 or request.request_has_state[-1].state_id == 4:
```

Comment:
Could you use the code instead of the ID?

Reply:
Sure.